managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -2151,7 +2151,18 @@ public void operationFailed(ManagedLedgerException exception) {
 
                 mdEntry.triggerFailed(exception);
             }
-        });
+        };
+
+        if (State.NoLedger.equals(STATE_UPDATER.get(this))) {
+            if (mdEntry.newPosition.equals(ledger.getLastConfirmedEntry())) {
```

Comment:
Please update the compare logic as we discussed to cover the case that LAC's entryId is `-1` and add a test to cover it

Reply:
Done